File: Namrata-Sanger/189. Rotate Array.js
/*
Day 16
189. Rotate Array
Link: https://leetcode.com/problems/rotate-array/
Level: Medium
*/

/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */

function reverse(nums,low,high){
        while(low < high){
            let temp = nums[low];
            nums[low] = nums[high];
            nums[high] = temp;
            low++;
            high--;
        }
}
    
var rotate = function(nums, k) {    
        while(k>nums.length){
            k -= nums.length;
        }
        let half = nums.length - k;
        reverse(nums,half,nums.length-1);
        reverse(nums,0,half-1);
        reverse(nums,0,nums.length - 1);
    
};

File: Ayush-Gupta/189. Rotate Array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        k %= n;
        reverse(nums.end() - k, nums.end()); //reverse nums[n-k] to nums[n-1]
        reverse(nums.begin(), nums.end() - k); //reverse nums[0] to nums[n-k-1]
        reverse(nums.begin(), nums.end());  //reverse complete array
    }
};